adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -1739,6 +1767,22 @@
                 }
             };
         }
+
+        function parseJSON(json_string, default_val) {
+            try {
+                return JSON.parse(json_string);
+            } catch (error) {
+                return default_val;
+            }
+        }
+
+        function createError(error, error_description) {
```

Comment:
Instead of making the error that is returned here a Object I'd much rather see specific instances of the `Error` class. I know that in some other places in the code this is already done, but I would consider it a bad practice.

For example:
```ts
promise.setError(new Error('Keycloak server is not reachable, most likely due to network problems.'));
```

Reply:
Hi, @jonkoops !
We can't use canonical JS Error objects now in JS Adapter because of:
- It will break contract specified in [keycloak.d.ts](https://github.com/keycloak/keycloak/blob/master/adapters/oidc/js/src/main/resources/keycloak.d.ts) TypeScript API, that states that in case of error KeycloakError should be returned
- @mhajas told that there is no place for now in JS Adapter where native Error objects used, and feared about compatibility issues.

Reply:
Ok, fair enough. Let's keep it like this then. I'll see if we can make this a breaking change at some point.